### PR TITLE
Updates for PTL package

### PR DESCRIPTION
**Pull request by mike0042**
#### Bug/feature Description
* PR #882 introduced the ability to build a pbspro-ptl RPM
* After it was merged, "make dist" failed to function properly

#### Affected Platform(s)
* Linux

#### Cause / Analysis / Design
* Wildcards are discouraged when using automake due to lack of portability
* The way the wildcards were expressed needed updating

#### Solution Description
* Updated the wildcard portions of the Makefiles
* Reordered some of the code in the spec file so that the PTL code is not so spread out
* Set ptl_prefix in the PBS_AC_ENABLE_PTL function and use it in the Makefiles

#### Testing logs/output
* Successfully built and installed both PBS Pro and PTL packages
* [ptl.log](https://github.com/PBSPro/pbspro/files/2681780/ptl.log)

#### Checklist:

- [X] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [X] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [X] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [X] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [X] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by hirenvadalia** [comment]
@mike0042 I think we should enable PTL RPM in Travis-CI, what you say?

**Comment by sandisamp** [comment]
@mike0042 Thanks for adding the wildcard function. I did try the same , although abandoned it when got the warning message. Something like this:

> test/fw/Makefile.am:44: warning: wildcard $(srcdir: non-POSIX variable name
> test/fw/Makefile.am:44: (probably a GNU make extension)

**Comment by mike0042** [comment]
@sandisamp : The warning is unavoidable if we choose to use wildcards. In their own words... https://www.gnu.org/software/automake/manual/html_node/Wildcards.html

Unless we list the files explicitly, we'll have to live with this warning. The piece I really need working is "make dist" which I use to create the package I hand off to rpmbuild. That is the proper "autotools" way of building PBS Pro.

**Comment by mike0042** [comment]
@PBSPro/pbspro-maintainers : Please consider this PR for merge. The tests were run using the pbspro-ptl RPM generated by the build.

**Comment by bhroam** [comment]
@mike0042 If wildcards are causing us warnings, why don't we list out the files like we do for PBS?

**Comment by mike0042** [comment]
@bhroam there was a brief discussion in PR #882 about the use of wildcards. The way it was implemented broke the "make dist" target because the $(wildcard ...) syntax was not used. Once added, the warnings appeared, but "make dist" worked again. I see both sides of the argument, whether to use wildcards or not. I think the problem is there will be many cases where people add tests and forget to update the Makefile.am. By using wildcards, they don't have to remember to update Makefile.am. I don't like warnings either (certainly cleaned up my share of them in the C code), but in this case I think the convenience outweighs being overly pedantic.

**Comment by bhroam** [comment]
@mike0042 You have a valid point.  We're adding tests all the time, and if we had to update one of the .ac files each time we did, that'd be a pain.  Not only that people would forget.  I agree that the wildcards are the way to go.  I saw more wildcards than just the tests though.  Could we spell out the files for the PTL lib and utils, but wildcard the tests?  Or is that not even worthwhile?

File: pbspro.spec [hunk 1]
```diff
@@ -543,7 +535,21 @@ fi
...
+%defattr(-,root,root, -)
+%dir %{ptl_prefix}
+%{ptl_prefix}/*
+%{_sysconfdir}/profile.d/ptl.csh
+%{_sysconfdir}/profile.d/ptl.sh
+%endif
+
 %changelog
+* Thu Dec 13 2018 Michael Karo <mike0042@gmail.com> - 1.26
+- Bring pbspro.spec.in and pbspro.spec back in sync
```
Comment (bhroam) [inline pbspro.spec]: I don't think this really belongs in the changelist for the spec file.  This isn't a change to the spec file, it is a change in the repo.  The change that is being made to the spec file is what was the original change.

File: pbspro.spec.in [hunk 2]
```diff
@@ -424,6 +425,22 @@ if [ "$1" != "1" ]; then
...
+fi
+
+%posttrans %{pbs_execution}
+# The %preun section of 14.x unconditially removes /etc/init.d/pbs
+# because it does not check whether the package is being removed
+# or upgraded. Make sure it exists here.
+if [ -r %{pbs_prefix}/libexec/pbs_init.d ]; then
+	install -D %{pbs_prefix}/libexec/pbs_init.d /etc/init.d/pbs
+fi
+
```
Comment (bhroam) [inline pbspro.spec.in]: Shouldn't this change also be seen in the generated spec file?

File: pbspro.spec.in [hunk 3]
```diff
@@ -527,7 +535,21 @@ fi
...
+%defattr(-,root,root, -)
+%dir %{ptl_prefix}
+%{ptl_prefix}/*
+%{_sysconfdir}/profile.d/ptl.csh
+%{_sysconfdir}/profile.d/ptl.sh
+%endif
+
 %changelog
+* Thu Dec 13 2018 Michael Karo <mike0042@gmail.com> - 1.26
+- Bring pbspro.spec.in and pbspro.spec back in sync
```
Comment (bhroam) [inline pbspro.spec.in]: As above, I don't think this should be in the changelog.

**Comment by mike0042** [comment]
@bhroam You also make a good point. We could do what you suggest since the test harness changes rather infrequently as compared to the tests themselves. However, we'd still have several warnings, one for each directory under test/tests. It's the C side of the code where we must be strict, and the PTL RPM isn't built by default.

File: pbspro.spec.in (hunk 2 above)
Comment (mike0042) [reply pbspro.spec.in]: It was already there. That's what brought them out of sync. The files are identical with the exception of the line where the version is defined. I manually ran a diff to ensure this was the case.

Comment (bhroam) [reply pbspro.spec.in]: OK cool, thanks

File: pbspro.spec (hunk 1 above)
Comment (bhroam) [reply pbspro.spec]: I like us having a changelog, but I think we should only put things in it that are changes to the spec file.  The fact that we have a pbspro.spec.in and pbspro.spec in our repo doesn't really have anything to do with the inner workings of the spec file itself.  I'd rather you take it out.  The real place for it is probably in your git log message if you were to provide more than one line.  We usually don't do that though.

**Comment by mike0042** [comment]
A quick update on the status of this PR. Some CentOS 7 systems are failing to start the database with these changes. It appears to be an issue with the shebang being /bin/bash rather than /bin/sh. We are exploring a solution.

**Comment by bhroam** [comment]
I merged in the PR.

**Pull request merged**